internal/nginx/config/generator.go

```diff
@@ -150,6 +174,60 @@ func generateProxyPass(address string) string {
 	return "http://" + address
 }
 
+var supportedFilters = map[v1beta1.HTTPRouteFilterType]struct{}{
+	v1beta1.HTTPRouteFilterRequestRedirect: {},
+}
+
+func findFirstFilters(
+	filters []v1beta1.HTTPRouteFilter,
+	filterTypes map[v1beta1.HTTPRouteFilterType]struct{},
+) map[v1beta1.HTTPRouteFilterType]v1beta1.HTTPRouteFilter {
+
+	result := make(map[v1beta1.HTTPRouteFilterType]v1beta1.HTTPRouteFilter)
+	for i := len(filters) - 1; i >= 0; i-- {
+		if _, exist := filterTypes[filters[i].Type]; exist {
+			result[filters[i].Type] = filters[i]
+		}
+	}
+
+	return result
+}
+
+func generateReturnValForRedirectFilter(filter *v1beta1.HTTPRequestRedirectFilter, listenerPort int) *returnVal {
+	if filter == nil {
+		return nil
+	}
+
+	hostname := "$host"
+	if filter.Hostname != nil {
+		hostname = string(*filter.Hostname)
+	}
+
+	// FIXME(pleshakov): Unknown values here must result in the implementation setting the Attached Condition for
+	// the Route to  `status: False`, with a Reason of `UnsupportedValue`. In that case, all routes of the Route will be
+	// ignored. NGINX will return 500. This should be implemented in the validation layer.
+	code := statusFound
+	if filter.StatusCode != nil {
+		code = statusCode(*filter.StatusCode)
+	}
+
+	port := listenerPort
+	if filter.Port != nil {
+		port = int(*filter.Port)
```

Comment:
If that happens, NGINX will generate a redirect to that port.
The Gateway API allows any valid port value.

Reply:
OK, but we only support ports 80 and 443. Do we want to reconfigure a redirect to a port that we know is not open on the Gateway?

Reply:
The redirect could be to a non-Gateway managed URL. like `https://random.nginx.org:8443/test`. In that case, the port could be any valid value.

However, the hostname could also belong to the Gateway.  An obvious case if when the hostname is not defined -- in that case, the hostname of the request is used, so we can assume it belongs to the Gateway. However, in general, not sure if there is an easy way to determine if a given redirect URL belongs to the Gateway. Because of that, and also because the spec doesn't say anything about port validation with regards to belonging to the Gateway, I don't think we need to do anything about it.